src/SprykerEco/Yves/Adyen/Theme/default/components/molecules/adyen-credit-card/adyen-credit-card.ts

```diff
@@ -38,20 +68,56 @@ export default class AdyenCreditCard extends Component {
             environment: this.environment,
             originKey: this.originKey,
             paymentMethodsResponse: JSON.parse(this.paymentMethodsResponse),
-            onChange: state => {
-                if (state.isValid) {
-                    this.cardNumberInput.value = state.data.paymentMethod.encryptedCardNumber;
-                    this.securityCodeInput.value = state.data.paymentMethod.encryptedSecurityCode;
-                    this.expiryYearInput.value = state.data.paymentMethod.encryptedExpiryYear;
-                    this.expiryMonthInput.value = state.data.paymentMethod.encryptedExpiryMonth;
-                }
-            }
+            onChange: (state: PaymentState) => {
+                this.onPaymentFormFieldsChange(state);
+            },
         };
 
         const checkout = new AdyenCheckout(configuration);
         checkout.create('card').mount(`#${this.containerId}`);
     }
 
+    protected onPaymentFormFieldsChange(state: PaymentState): void {
+        const paymentMethod = state.data.paymentMethod;
+        const encryptedData = [
+            paymentMethod.encryptedCardNumber,
+            paymentMethod.encryptedSecurityCode,
+            paymentMethod.encryptedExpiryYear,
+            paymentMethod.encryptedExpiryMonth,
+        ];
+
+        this.isFormValid = state.isValid;
+        this.toggleSubmitButtonDisabling(!this.isFormValid);
+        this.isFormValid ? this.fillFormHiddenFields(encryptedData) : this.fillFormHiddenFields(['', '', '', '']);
+    }
+
+    protected fillFormHiddenFields([cardNumber, securityCode, expiryYear, expiryMonth]: string[]): void {
```

Comment:
Please use default parameters for `fillFormHiddenFields` instead of `this.fillFormHiddenFields(['', '', '', ''])`